### ISSUE TITLE
Ability to upload guest difficulties as .osu files onto the osu website.

### ISSUE THREAD

**Issue by SecreOsu** [ppy/osu-web#7321]
First github post so sorry if formatting is bad, but this kind of ties with the request posted https://github.com/ppy/osu-web/issues/2165 in that support for guest difficulties right now is currently a little lackluster.

As of right now the only proper way to do guest difficulties for others is to send .osu files through discord or sending the whole .osu text through a pm. 

What I suggest is the ability to upload a .osu file (possibly through osu pms or through a beatmap's page) that the mapset host then could take download directly into their mapset. If the ability to upload a .osu file through a beatmaps page was implemented, being able to download that .osu directly into the .osz for the beatmapset would greatly help organizing guest difficulties for all mappers.

Considering a .osu is just a text file, it shouldn't be that much of an issue regarding uploading the actual file (possibly through the use of a drag/drop or attach system). 

tldr; .osu upload system for beatmap guest difficulties, ideally the download would download the .osu straight into the osz so you would not need to search for the .osz every time you want to upload a gder's guest difficulty. This also allows for a more streamlined approach towards guest difficulties in general.

**Comment by dorothy3242** [comment]
So for a summary of yesterday's discussion in modding

@peppy Is considering adding a hack like so->
adding text to the description such as 
`+GD: Noffy` and that enabling the user to update the diffs through BSS themselves.

Adding in full support may require too much overhead ATM but he did say this kind of hack is likely feasible within a short timeframe. peppy correct me if I'm wrong. Trying to summarize so it's noted down on the issue tracker.

**Comment by peppy** [comment]
Going to close this in favour of https://github.com/ppy/osu-web/issues/2165.

Note that even though that is an old issue, I will be tracking it and hope to have something in place over the next month.
